### PR TITLE
#67: Implement report command and other small fixes

### LINKED ISSUE #67: Command for reporting a user to the moderators
In the case that someone breaks the server rules, users should be able to report specific persons to the moderators. This could be done via a command like `!report [User]` where you should be able to simply write a users nickname instead of direlcty mentioning him. The second could trigger a notification which will then be sent to the offender, which results in him/her knowing, who informed the moderators.

To inform the moderators, an embed will be posted in a configured report channel on the server.

### PR DESCRIPTION
**Pull request by PKlempe**
Closes #67

### CODE REVIEW

File: bot/admin/admin.py [hunk 1]
```diff
@@ -41,14 +41,14 @@ async def echo(self, ctx: commands.Context, channel: Optional[discord.TextChanne
 
     @commands.group(name='embed', hidden=True, invoke_without_command=True)
     @command_log
-    async def embed(self, ctx: commands.Context, channel: discord.TextChannel, color: discord.Colour, *, text: str):
+    async def embed(self, _ctx: commands.Context, channel: discord.TextChannel, color: discord.Colour, *, text: str):
```
Comment (SebastiantheAstorian) [inline bot/admin/admin.py:44]: Does mypy not say anything anymore. 
I actually thought about something like: 
`async def embed(self, _: commands.Context, channel: discord.TextChannel, color: discord.Colour, *, text: str):`

File: bot/moderation/moderation.py [hunk 2]
```diff
@@ -24,10 +25,48 @@ def __init__(self, bot):
...
+
+        Handles specific exceptions which occur during the execution of this command. The global error handler is still
+        being called for every error thrown.
+
+        Args:
+            ctx (discord.ext.commands.Context): The context in which the command was called.
+            error (commands.CommandError): The error raised during the execution of the command.
+        """
+        if isinstance(error, commands.BadArgument):
+            regex = re.search("\"(.*)\"", error.args[0])
```
Comment (SebastiantheAstorian) [inline bot/moderation/moderation.py:61]: What does this regex search for (hard to read in code)? maybe a short inline comment?

File: bot/university/university.py [hunk 3]
```diff
@@ -83,8 +83,8 @@ async def ufind_get_staff_data(self, ctx: commands.Context, *, search_term: str)
     async def ufind_error(self, ctx, error: commands.CommandError):
         """Error Handler for the `ufind` subcommand `staff`.
 
-        Handles specific exceptions which occur during the execution of this command. Other exceptions will be simply
-        re-raised so that they can be handled somewhere else.
+        Handles specific exceptions which occur during the execution of this command. The global error handler is still
+        being called for every error thrown.
```
Comment (SebastiantheAstorian) [inline bot/university/university.py:87]: The global error handler is still called for every error thrown.

File: bot/admin/admin.py (hunk 1 above)
Comment (PKlempe) [reply bot/admin/admin.py:44]: Nope, MyPy is fully ok with this and doesn't whine about that this argument isn't being used anymore.

File: bot/moderation/moderation.py (hunk 2 above)
Comment (PKlempe) [reply bot/moderation/moderation.py:61]: Get the text between two quotes.

File: bot/moderation/moderation.py [hunk 4]
```diff
@@ -50,15 +50,15 @@ async def report_user(self, ctx: commands.Context, offender: discord.Member, *,
...
+        Handles specific exceptions which occur during the execution of this command. The global error handler will
+        still be called for every error thrown.
 
         Args:
             ctx (discord.ext.commands.Context): The context in which the command was called.
             error (commands.CommandError): The error raised during the execution of the command.
         """
         if isinstance(error, commands.BadArgument):
-            regex = re.search("\"(.*)\"", error.args[0])
+            regex = re.search("\"(.*)\"", error.args[0])  # Regex to get the text between two quotes.
```
Comment (SebastiantheAstorian) [inline bot/moderation/moderation.py:61]: could use raw string

**Pull request merged**